browser/ui/views/split_view/split_view.cc

```diff
@@ -294,96 +273,47 @@ void SplitView::OnDidBreakTile(const TabTile& tile) {
   UpdateContentsWebViewVisual();
 }
 
-void SplitView::OnSwapTabsInTile(const TabTile& tile) {
-  if (!IsActiveWebContentsTiled(tile)) {
-    return;
-  }
-
-  UpdateSecondaryContentsWebViewVisibility();
-}
-
-tabs::TabHandle SplitView::GetActiveTabHandle() const {
-  auto* model = browser_->tab_strip_model();
-  if (model->empty()) {
-    return {};
-  }
-  return model->GetTabAtIndex(model->active_index())->GetHandle();
-}
-
-bool SplitView::IsActiveWebContentsTiled(const TabTile& tile) const {
-  auto active_tab_handle = GetActiveTabHandle();
-  return tile.first == active_tab_handle || tile.second == active_tab_handle;
-}
-
-bool SplitView::IsWebContentsTiled(content::WebContents* contents) const {
-  const int tab_index =
-      browser_->tab_strip_model()->GetIndexOfWebContents(contents);
-  if (tab_index == TabStripModel::kNoTab) {
-    return false;
-  }
-  const auto tab_handle =
-      browser_->tab_strip_model()->GetTabAtIndex(tab_index)->GetHandle();
-  return browser_->GetFeatures().split_view_browser_data()->IsTabTiled(
-      tab_handle);
-}
-
 void SplitView::UpdateSplitViewSizeDelta(content::WebContents* old_contents,
                                          content::WebContents* new_contents) {
-  auto get_index_of = [this](content::WebContents* contents) {
-    return browser_->tab_strip_model()->GetIndexOfWebContents(contents);
-  };
-  if (get_index_of(old_contents) == TabStripModel::kNoTab ||
-      get_index_of(new_contents) == TabStripModel::kNoTab) {
+  auto* model = browser_->tab_strip_model();
+  if (model->GetIndexOfWebContents(old_contents) == TabStripModel::kNoTab ||
+      model->GetIndexOfWebContents(new_contents) == TabStripModel::kNoTab) {
     // This can happen on start-up or closing a tab.
     return;
   }
 
-  auto* split_view_browser_data =
-      browser_->GetFeatures().split_view_browser_data();
-  auto get_tab_handle = [this, &get_index_of](content::WebContents* contents) {
-    return browser_->tab_strip_model()
-        ->GetTabAtIndex(get_index_of(contents))
-        ->GetHandle();
-  };
-  auto old_tab_handle = get_tab_handle(old_contents);
-  auto new_tab_handle = get_tab_handle(new_contents);
-
-  auto old_tab_tile = split_view_browser_data->GetTile(old_tab_handle);
-  auto new_tab_tile = split_view_browser_data->GetTile(new_tab_handle);
-  if ((!old_tab_tile && !new_tab_tile) || old_tab_tile == new_tab_tile) {
-    // Both tabs are not tiled, or in a same tile. So we don't need to update
-    // size delta
+  auto* controller = browser_->GetFeatures().split_view_controller();
+  // Both tabs are not related with split view, or shown together.
+  // So we don't need to update size delta.
+  if (!controller->IsOpenedFor(old_contents) &&
+      !controller->IsOpenedFor(new_contents)) {
+    return;
+  }
+  if (controller->AreShowingTogether({old_contents, new_contents})) {
```

Comment:
so we only need to update the size delta if:
1. Both tabs are tiled
2. They aren't in the same tile (as the delta will be the same)